src/hotspot/share/opto/library_call.cpp

```diff
@@ -4799,6 +4799,18 @@ bool LibraryCallKit::inline_native_hashcode(bool is_virtual, bool is_static) {
     return true;
   }
 
+  const TypeInstPtr* t = _gvn.type(obj)->isa_instptr();
+  if (t != nullptr && t->const_oop() != nullptr) {
+    assert(!is_virtual, "no devirtualization for constant receiver?");
+    ciConstant identity_hash = t->const_oop()->identity_hash();
+    if (identity_hash.is_valid()) {
```

Comment:
Is there any reason we don't calculate the identity hash right away if there is not any?